src/test/java/com/example/demo/WebClientTests.java

```diff
@@ -0,0 +1,57 @@
+package com.example.demo;
+
+import com.example.demo.client.JsonPlaceholderWebClient;
+import com.example.demo.client.PostObject;
+import com.example.demo.smokeSpecSuite.BaseSpec;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+@SpringBootTest(classes = WebClientTests.class)
```

Comment:
you can attach this to the base spec only

Reply:
is this test runs as spring boot application?

Reply:
This annotation is used to start the Spring context for integration tests. This will bring up the full autoconfigruation context.